src/InfiniteIterableValue.php

```diff
@@ -20,6 +20,21 @@ public function __construct(iterable $iterable)
         $this->stack = new IterableValueStack(new IterableValueIterator($iterable));
     }
 
+    /**
+     * @template TNewKey
+     * @template TNewValue
+     * @phpstan-param IterableValueStack<TNewKey, TNewValue> $stack
+     * @phpstan-return self<TNewKey, TNewValue>
+     */
+    private function fromStack(IterableValueStack $stack): self
```

Comment:
Kind of constructor, so I guess it could be `public static`

Reply:
Ok, I've made it static, but not public. `IterableValueStack` should not leak outside